### ISSUE TITLE
Memory Leaks

### ISSUE THREAD

**Issue by yorrd** [Dabolus/lit-rx#1]
You're subscribing to observables, but they're not being unsubscribed when the part is removed. Doesn't this create memory leaks?

**Comment by Dabolus** [comment]
@yorrd yes, definitely. This will be fixed when (if) we get [a disconnected handler for directives](https://github.com/Polymer/lit-html/issues/283). Besides that, I just noticed that there is also a memory leak when an observable gets replaced with another, since the old observables aren't currently being unsubscribed. I'm going to fix it in the next couple of days.

**Comment by yorrd** [comment]
@Dabolus great. I'm active in that issue as well, was just wondering if you knew of that. Looking forward to where this is going. Currently, we solve the memory leak issues by extending lit-html with unmount handlers. That's not abstracted away though atm, one part is monkey-patched because the respective API is not public

**Comment by tyroneneill** [comment]
@Dabolus @yorrd while we wait for the handlers to be implemented in the next version of ```lit-html``` I think we can safely handle this ourselves. 

I have had a go at it [here](https://github.com/tyroneneill/lit-study/blob/master/src/my-element.ts), have not added any test coverage yet but can confirm the destroy hooks are activated if we remove the bound dom element.  

It does come with a **trade-off** in that the custom element needs to use a corresponding mixin: 

```typescript
@customElement('my-element')
export class MyElement extends LifecycleMixin(LitElement) {
   ...
}
```

I don't really want to create/manage a library (lazy) and thinking maybe I could raise a PR against ```lit-rx```?

**Comment by yorrd** [comment]
@tyroneneill I've implemented this in our stack, but it's very deep inside and not well abstracted away. Looking forward to lit 3.0 to finally fix this the right way

**Comment by klauss194** [comment]
hey guys, was curious if there is any update on this matter 
I'm loving Rx so far. was thinking how would an Rx & lit app look like ?

**Comment by yorrd** [comment]
@klauss194 at the moment, we're using a custom controller (we call it RXController) that subscribes and unsubscribes on connect / disconnect. You pass an observable and the value is exposed via `.value`. With the new lit versions, this is pretty straight forward.
